### ISSUE TITLE
pytest-bdd 3.0.0 does not seem to work with pytest 4.0.1

### ISSUE THREAD

**Issue by Vic152** [pytest-dev/pytest-bdd#283]
Hey Guys,

pytest-bdd 3.0.0 does not seem to work with pytest 4.0.1.

What is the highest version of pytest that pytest-bdd supports?

I am running with pytest 3.9.1 now. I use python 3.5.4.

**Comment by sliwinski-milosz** [comment]
Please share the error message/exception that you can see when you use pytest-bdd 3.0.0 with pytest 4.0.1

**Comment by The-Compiler** [comment]
Works fine here FWIW.

**Comment by sliwinski-milosz** [comment]
I have tried it on some test. This is what I got:
```python
tests/test_simple.py:5:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/scenario.py:154: in _execute_scenario
    scenario=scenario,
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/hooks.py:284: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/manager.py:67: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/manager.py:61: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/plugin.py:67: in pytest_bdd_before_scenario
    reporting.before_scenario(request, feature, scenario)
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/reporting.py:167: in before_scenario
    request.node.__scenario_report__ = ScenarioReport(scenario=scenario, node=request.node)
../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/reporting.py:78: in __init__
    param_names = parametrize.args[0] if isinstance(parametrize.args[0], (tuple, list)) else [
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MarkInfo Mark(name='parametrize', args=(['some_query'], [['test']]), kwargs={})>

    def warned(self):
>       warnings.warn(warning, stacklevel=2)
E       _pytest.warning_types.RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
E       Please use node.get_closest_marker(name) or node.iter_markers(name).
E       Docs: https://docs.pytest.org/en/latest/mark.html#updating-code

../../../.virtualenvs/some-tests/lib/python3.5/site-packages/_pytest/mark/structures.py:25: RemovedInPytest4Warning
```

So indeed it doesn't work with pytest 4.0.1 for parametrized tests.

I tried above test with pytest-bdd from master branch and it works fine.
Tests for version 3.0.1 (not released yet) pass so most likely that version will work with latest pytest.
Thank you for your patience.

**Comment by Vic152** [comment]
> I have tried it on some test. This is what I got:
> 
> ```python
> tests/test_simple.py:5:
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/scenario.py:154: in _execute_scenario
>     scenario=scenario,
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/hooks.py:284: in __call__
>     return self._hookexec(self, self.get_hookimpls(), kwargs)
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/manager.py:67: in _hookexec
>     return self._inner_hookexec(hook, methods, kwargs)
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pluggy/manager.py:61: in <lambda>
>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/plugin.py:67: in pytest_bdd_before_scenario
>     reporting.before_scenario(request, feature, scenario)
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/reporting.py:167: in before_scenario
>     request.node.__scenario_report__ = ScenarioReport(scenario=scenario, node=request.node)
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/pytest_bdd/reporting.py:78: in __init__
>     param_names = parametrize.args[0] if isinstance(parametrize.args[0], (tuple, list)) else [
> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
> 
> self = <MarkInfo Mark(name='parametrize', args=(['some_query'], [['test']]), kwargs={})>
> 
>     def warned(self):
> >       warnings.warn(warning, stacklevel=2)
> E       _pytest.warning_types.RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
> E       Please use node.get_closest_marker(name) or node.iter_markers(name).
> E       Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
> 
> ../../../.virtualenvs/some-tests/lib/python3.5/site-packages/_pytest/mark/structures.py:25: RemovedInPytest4Warning
> ```
> 
> So indeed it doesn't work with pytest 4.0.1 for parametrized tests.
> 
> I tried above test with pytest-bdd from master branch and it works fine.
> Tests for version 3.0.1 (not released yet) pass so most likely that version will work with latest pytest.
> 
> Version 3.0.1 will be released as soon as it will be fully tested and ready. Thank you for your patience.

Yes, this indeed was the error. Sorry :) I was too slow to submit the stack trace.

**Comment by youtux** [comment]
I just released pytest-bdd 3.0.1 on PyPI, sorry for the long wait.
